dev/tests/integration/testsuite/Magento/Catalog/Model/Product/UrlTest.php

```diff
@@ -42,6 +42,51 @@ public function testGetUrlInStore()
         $this->assertStringEndsWith('simple-product.html', $this->_model->getUrlInStore($product));
     }
 
+    /**
+     * @magentoDataFixture Magento/Store/_files/second_store.php
+     * @magentoConfigFixture default_store web/unsecure/base_url http://sample.com/
+     * @magentoConfigFixture default_store web/unsecure/base_link_url http://sample.com/
+     * @magentoConfigFixture fixturestore_store web/unsecure/base_url http://sample-second.com/
+     * @magentoConfigFixture fixturestore_store web/unsecure/base_link_url http://sample-second.com/
+     * @magentoDataFixture Magento/Catalog/_files/product_simple_multistore.php
+     * @magentoAppArea adminhtml
+     */
+    public function testGetUrlInStoreWithSecondStore()
+    {
+        $repository = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+            \Magento\Catalog\Model\ProductRepository::class
+        );
+        /** @var \Magento\Store\Model\Store $store */
+        $store = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+            ->create(\Magento\Store\Model\Store::class);
+        $store->load('fixturestore', 'code');
+        /** @var \Magento\Store\Model\Store $store */
+        $storeDefault = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+            ->create(\Magento\Store\Model\Store::class);
+        $storeDefault->load('default', 'code');
+
+        $product = $repository->get('simple');
+        $expected = $this->testGetUrlsWithSecondStoreProvider();
+
+        $this->assertEquals(
+            $expected['fixturestore'],
+            $this->_model->getUrlInStore($product, ['_scope'=> $store->getId(), '_nosid' => true])
+        );
+
+        $this->assertEquals(
+            $expected['default'],
+            $this->_model->getUrlInStore($product, ['_scope' => $storeDefault->getId(), '_nosid' => true])
+        );
+    }
+
+    public function testGetUrlsWithSecondStoreProvider()
+    {
+        return [
```

Comment:
the correct return format to have 2 arguments ($storeCode, $expectedProductUrl) in the test is:

    return ['iteration1' => ['fixturestore', 'http://...'], 'interation2' => ['default', 'http://...']];